flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSinkFactory.java

```diff
@@ -0,0 +1,125 @@
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.CONNECT_TIMEOUT;
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.DRIVER_CLASS_NAME;
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.HOSTNAME;
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.PASSWORD;
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.PORT;
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.SERVER_TIME_ZONE;
+import static org.apache.flink.cdc.connectors.mysql.sink.MySqlDataSinkOptions.USERNAME;
+
+/** Factory for creating configured instance of {@link DataSinkFactory}. */
+public class MySqlDataSinkFactory implements DataSinkFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(MySqlDataSinkFactory.class);
+
+    public static final String IDENTIFIER = "mysql-writer";
+
+    @Override
+    public DataSink createDataSink(Context context) {
+        FactoryHelper.createFactoryHelper(this, context)
+                .validateExcept(JDBC_PROPERTIES_PROP_PREFIX);
+
+        final Configuration config = context.getFactoryConfiguration();
+        MySqlDataSinkConfig.Builder builder = new MySqlDataSinkConfig.Builder();
+
+        config.getOptional(HOSTNAME).ifPresent(builder::hostname);
+        config.getOptional(PORT).ifPresent(builder::port);
+        config.getOptional(USERNAME).ifPresent(builder::username);
+        config.getOptional(PASSWORD).ifPresent(builder::password);
+        builder.serverTimeZone(config.getOptional(SERVER_TIME_ZONE).orElseGet(() -> "UTC"));
+        builder.connectTimeout(
+                config.getOptional(CONNECT_TIMEOUT).orElseGet(() -> Duration.ofSeconds(30)));
+        builder.connectionPoolSize(config.getOptional(CONNECTION_POOL_SIZE).orElseGet(() -> 20));
+        builder.connectMaxRetries(config.getOptional(CONNECT_MAX_RETRIES).orElseGet(() -> 3));
+        // driver class name
+        builder.driverClassName(
+                config.getOptional(DRIVER_CLASS_NAME).orElseGet(() -> "com.mysql.cj.jdbc.Driver"));
+        // builder jdbc properties
+        Properties properties = new Properties();
+        Map<String, String> jdbcProperties =
+                JdbcSinkOptions.getPropertiesByPrefix(config, JDBC_PROPERTIES_PROP_PREFIX);
+        properties.putAll(jdbcProperties);
+        builder.jdbcProperties(properties);
+        // get jdbc url
+        String jdbcUrl = MysqlPooledDataSinkFactory.INSTANCE.getJdbcUrl(builder.build());
+        builder.connUrl(jdbcUrl);
+        // print configs
+        Map<String, String> map = config.toMap();
+        OptionUtils.printOptions(IDENTIFIER, map);
+
+        MySqlDataSinkConfig sinkConfig = builder.build();
+        MySqlCatalog catalog = MySqlCatalogFactory.INSTANCE.createCatalog(sinkConfig);
+
+        return new JdbcDataSink(catalog, sinkConfig);
```

Comment:
Can we add an ItTest to write event using this JDBCDataSink and verify the result to make sure that it can work as expected?